src/Core/Framework/Api/Controller/SalesChannelProxyController.php

```diff
@@ -135,6 +141,56 @@ public function assignCustomer(Request $request, Context $context): Response
         return $response;
     }
 
+    #[Route(path: '/api/_proxy/login-as-customer-token-generate', name: 'api.proxy.login-as-customer-token-generate', methods: ['POST'], defaults: ['_acl' => ['api_proxy_login-as-customer']])]
+    public function loginAsCustomerTokenGenerate(Request $request, Context $context): Response
+    {
+        if (!$request->request->has(self::SALES_CHANNEL_ID)) {
+            throw ApiException::salesChannelIdParameterIsMissing();
+        }
+
+        if (!$request->request->has(self::CUSTOMER_ID)) {
+            throw ApiException::customerIdParameterIsMissing();
+        }
+
+        $customerId = $request->request->getString('customerId');
+        if (!$this->doesCustomerExist($customerId, $context)) {
+            throw ApiException::resourceNotFound('customer', ['id' => $customerId]);
+        }
```

Comment:
Can you not use the data validator like in your StoreApiRoute to validate this?

Reply:
Used the Validator in https://github.com/shopware/shopware/pull/3727/commits/65ddefacfe73f7c590d4acfed239fc8279cc16c4